### PR TITLE
Fix apparent corner case with `import/no-commonjs` rule

### PR DESCRIPTION
**Pull request by charlessuh**
`var x = require("x")` will throw an error, but `var x; x = require("x")` does not.

### CODE REVIEW

File: tests/src/rules/no-commonjs.js [hunk 1]
```diff
@@ -28,27 +28,27 @@ ruleTester.run('no-commonjs', require('rules/no-commonjs'), {
...
-    { code: 'var a = c && require("b")' }, // conditional requires allowed
-    { code: 'require.resolve("help")' }, // methods of require are allowed
-    { code: 'require.ensure([])' }, // webpack specific require.ensure is allowed
-    { code: 'require([], function(a, b, c) {})' }, // AMD require is allowed
-    { code: "var bar = require('./bar', true);" },
-    { code: "var bar = proxyquire('./bar');" },
-    { code: "var bar = require('./ba' + 'r');" },
-    { code: 'var zero = require(0);' },
-    { code: 'require("x")', options: [{ allowRequire: true }] },
+    { code: 'function a() { var x = require("y"); }', parserOptions: { sourceType: 'module' } }, // nested requires allowed
```
Comment (ljharb) [inline tests/src/rules/no-commonjs.js]: why are all these parserOptions required?

Comment (charlessuh) [reply tests/src/rules/no-commonjs.js]: This [line](https://github.com/benmosher/eslint-plugin-import/blob/5101b73effbf3706495f62121a8719f0ea0e2c68/src/rules/no-commonjs.js#L90) causes the check for `require` to only occur in a module scope.

**Comment by ljharb** [comment]
I think this can be called a bugfix, even though it will cause more warnings to be issued.
